### PR TITLE
Fix: Global Colors - Gradient control doesn't work on frontend [ED-3517] (Fix #13288)

### LINKED ISSUE #13288: Bug with global values in linear-gradient css function
This rule (applied on a color control):
```php
'selectors' => [
    '{{WRAPPER}} .elementor-button::after' => 'background-image: linear-gradient(to right {{color_from_another_control.VALUE}} 50%, {{VALUE}} 50%)',
],
```
when the color values involved are hex values (from the color-picker) correctly outputs:
```css
background-image: linear-gradient(to right, #FF0000 50%, #558ECC 50%);
```
 but breaks with custom css vars (i.e when a global color is selected) and outputs something like this:
```css
background-image: var( --e-global-color-accent_1 );
```  

Expected output would be:
```css
background-image: linear-gradient(to right, var( --e-global-color-accent_1 ) 50%, var( --e-global-color-accent_2 ) 50%);
```

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix: Gradient control doesn't work on frontend when using Global Colors (Fixes #13288)

### CODE REVIEW

File: core/files/css/base.php [hunk 1]
```diff
@@ -673,6 +678,38 @@ protected function add_control_style_rules( array $control, array $values, array
...
+	 * fetches and returns the global default value. Otherwise, it returns null.
+	 *
+	 * @since 3.7.0
+	 * @access private
+	 *
+	 * @param $control
+	 * @return string|null
+	 */
+	private function get_control_global_default_value( $control ) {
+		if ( ! empty( $control['global']['default'] ) ) {
```
Comment (KingYes) [inline core/files/css/base.php:694]: Try `if ( empty( $control['global']['default'] ) ) {` and exit from the function early..

File: core/files/css/base.php [hunk 2]
```diff
@@ -673,6 +678,38 @@ protected function add_control_style_rules( array $control, array $values, array
...
+			// If the control value is empty, and the control has a global default set, fetch the global value and use it.
+			$global_enabled = false;
```
Comment (KingYes) [inline core/files/css/base.php:696]: What's the different between `false` to `null` here?

**Review by KingYes — CHANGES_REQUESTED** [review]

File: core/files/css/base.php (hunk 2 above)
Comment (udidol) [reply core/files/css/base.php]: The difference is that the following methods return booleans that override this variable's value. So I prefer to instantiate it as a boolean as well to keep the typing identical.

File: core/files/css/base.php (hunk 1 above)
Comment (udidol) [reply core/files/css/base.php:694]: Done

File: core/files/css/base.php [hunk 3]
```diff
@@ -442,6 +442,11 @@ public function parse_property_placeholder( array $control, $value, array $contr
 			$value = call_user_func( $value_callback, $control );
 		}
 
+		// If the control value is empty, check for global default.
```
Comment (matipojo) [inline core/files/css/base.php:445]: Please add a screenshot test

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/screenshotter/config/buttons.json [hunk 4]
```diff
@@ -1 +1 @@
...
...lements":[],"isInner":false,"widgetType":"text-editor","elType":"widget"},{"id":"502b7d2d","settings":{"text":"Gradient background","background_background":"gradient"},"elements":[],"isInner":false,"widgetType":"button","elType":"widget"},{"id":"4b0a1e2f","settings":{"text":"Radius","background_color":"#000000","border_radius":{"unit":"px","top":"50","right":"50","bottom":"50","left":"50","isLinked":true}},"elements":[],"isInner":false,"widgetType":"button","elType":"widget"},{"id":"cc111f6","settings":{"text":"Shadow","background_color":"#000000","button_box_shadow_box_shadow_type":"yes","button_box_shadow_box_shadow":{"horizontal":0,"vertical":0,"blur":10,"spread":5,"color":"rgba(0,0,0,0.5)"}},"elements":[],"isInner":false,"widgetType":"button","elType":"widget"},{"id":"774dc81b","settings":{"text":"Padding","background_color":"#000000","button_box_shadow_box_shadow":{"horizontal":0,"vertical":0,"blur":10,"spread":5,"color":"rgba(0,0,0,0.5)"},"text_padding":{"unit":"px","top":"30","right":"30","bottom":"30","left":"30","isLinked":true}},"elements":[],"isInner":false,"widgetType":"button","elType":"widget"}],"isInner":false,"elType":"column"}],"isInner":false,"elType":"section"}]}
```
Comment (matipojo) [inline tests/screenshotter/config/buttons.json:1]: Please move to global setting test

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/screenshotter/config/buttons.json (hunk 4 above)
Comment (udidol) [reply tests/screenshotter/config/buttons.json]: Done.

**Pull request merged**